### PR TITLE
[improve](distinct agg) add check of hash table to decide whether emplace value

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/exec/distinct_vaggregation_node.cpp [hunk 1]
```diff
@@ -57,24 +59,40 @@ Status DistinctAggregationNode::_distinct_pre_agg_with_serialized_key(
...
     SCOPED_TIMER(_insert_keys_to_column_timer);
     bool mem_reuse = _make_nullable_keys.empty() && out_block->mem_reuse();
     if (mem_reuse) {
         for (int i = 0; i < key_size; ++i) {
-            auto dst = out_block->get_by_position(i).column->assume_mutable();
-            key_columns[i]->append_data_by_selector(dst, _distinct_row);
+            auto output_column = out_block->get_by_position(i).column;
+            if (_stop_emplace_flag) { // swap the column directly, to solve Check failed: d.column->use_count() == 1 (2 vs. 1)
+                out_block->replace_by_position(i, key_columns[i]->assume_mutable());
+                in_block->replace_by_position(result_idxs[i], output_column);
```
Comment (HappenLee) [inline be/src/vec/exec/distinct_vaggregation_node.cpp:71]: clone_reisze()

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**